ls1intum/Artemis
# `Programming exercises`: Allow editing submission policy only on the grading page

### Checklist
#### General


#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


Resolves [#6497](https://github.com/ls1intum/Artemis/issues/6497)
When adding a submission policy with a submission limit to a programming exercise that previously had none, an error is encountered. Similarly, when updating an existing submission policy, no changes are made to the database and no error messages are shown.

### Description

Editting submission policy and submisson limit over programming exercise edit page is disabled with the banner stating that it can be done on the grading page (as stated in the docs: https://docs.artemis.cit.tum.de/user/exercises/programming/#updating-submission-policies). Also the banner is made visible for none policy option as well.  

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a programming exercise without a submission policy
2. Navigate to edit programming exercise
3. Verify submission policy is disabled

### Review Progress






#### Performance Review

#### Code Review
#### Manual Tests


### Test Coverage







### Screenshots


Screenshot 1 (Submission Policy: None):
![image](https://github.com/ls1intum/Artemis/assets/11020241/51ea2047-dd11-4c77-a5c9-3732041c5b2d)

Screenshot 2 (Submission Policy: Lock Repository)
![image](https://github.com/ls1intum/Artemis/assets/11020241/05b0dd8a-1c3c-43fb-860b-d8120a511bc9)

Screenshot 3 (Submission Policy: Submission Penalty)
![image](https://github.com/ls1intum/Artemis/assets/11020241/a8797318-a0ac-4209-91ed-3af22be7103b)
